### ISSUE TITLE
Doesn't work?

### ISSUE THREAD

**Issue by rico345100** [gajus/eslint-plugin-flowtype#369]
I followed the instructions in readme.md, however still ESLint gives me errors in Flow syntax:

```javascript
module.exports = {
  extends: [
    'airbnb',
    'plugin:flowtype/recommended'
  ],
  parser: 'babel-eslint',
...
rules: {
    // Flow Rules
    "flowtype/define-flow-type": 1,
    "flowtype/use-flow-type": 1,
    "flowtype/valid-syntax": 1,
    // ESLint Rules
    'arrow-parens': ['error', 'as-needed'],
    'comma-dangle': ['error', 'only-multiline'],
...
}
```

```javascript
export default class Transaction extends Component<ITransactionProps, ITransactionState> {
  // Type Defs - Properties
  newTabIndex: number;  // [eslint]: newTabIndex should be placed after onEdit

...
```

newTabIndex is not variable declaration, it's type declaration which Flow requires before using it. But still ESLint gives me above error.

**Comment by pnevyk** [comment]
It seems that you have a rule enabled which checks if class properties are sorted. And I suppose you have a `newTabIndex` above `onEdit` so they are not in alphabetical order. Note that although you use it just as Flow type declarations, the following syntax by [this](https://github.com/tc39/proposal-class-fields#field-declarations) spec might be also valid in the future (and it is now already supported by babel [plugin](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties)):

```js
class Foo {
    foo = 'bar';
}
```

So these class properties are not used just for Flow annotations but will be a syntax of JavaScript itself, so it doesn't surprise me that some rule for alphabetical order checking also checks class properties. I hope I explained it well.

**Comment by rico345100** [comment]
Thank you. Actually it was worked, I thought that ESLint misunderstand type definition to value declaration, but it wasn't. It was happened from somewhere else(maybe react related plugin in ESLint? or this? don't know), and fixed by put this in eslintrc:

```javascript
'react/sort-comp': [1, {
  order: [
    'type-annotations',    // THIS FIXED MY ERROR
    'static-methods',
    'lifecycle',
    'everything-else',
    'render'
  ]
}],
```

Anyway thank you for answer!

**Comment by pnevyk** [comment]
Now it is fully explained. In Airbnb eslint config which you are using there is [this](https://github.com/airbnb/javascript/blob/9a5267684b59106523cf5419439fd0d53c0a512e/packages/eslint-config-airbnb/rules/react.js#L233-L244) setting for `react/sort-comp`. As you can see, they don't have `type-annotations` in the order specification so it falls into `everything-else` group. However, above `everything-else` there is group with `/^on.+$/` pattern that matches your onEdit method and that's why eslint complained.
